contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -375,11 +390,18 @@ abstract contract BaseBalancerStrategy is InitializableAbstractStrategy {
         internal
         returns (uint256 unwrappedAmount)
     {
-        if (asset == stETH) {
+        if (asset == stETH && amount > 0) {
+            // amount is amount of wstETH to be unwrapped as stETH
             unwrappedAmount = IWstETH(wstETH).unwrap(amount);
-        } else if (asset == frxETH) {
-            unwrappedAmount = IERC4626(sfrxETH).withdraw(
-                amount,
+        } else if (asset == frxETH && amount > 0) {
+            /* redeem takes amount parameter that are the number of
+             * shares (sfrxETH) to be exchanged for assets (frxETH)
+             */
+            unwrappedAmount = IERC4626(sfrxETH).redeem(
+                /* adding + 1 here since a rounding error can 
+                 * return value that is off by 1 WEI
+                 */
+                amount + FRX_ETH_REDEEM_CORRECTION,
```

Comment:
nit: I'm just curious, how are other platforms/protocols dealing with this issue?

Reply:
I have yet to investigate that. Being 1 WEI off is something we've seen with balancer as well. Protocols usually round in the direction that serves the protocol that owns the code. To mitigate flash loan attacks as best as possible.